web/src/stream_topic_history.ts

```diff
@@ -46,21 +67,49 @@ export function all_topics_in_cache(sub: StreamSubscription): boolean {
         return true;
     }
 
-    // Now, we can just compare the first cached message to the first
-    // message ID in the stream; if it's older, we're good, otherwise,
-    // we might be missing the oldest topics in this stream in our
-    // cache.
     const first_cached_message = all_messages_data.first();
```

Comment:
This was a preexisting bug, but I think this should be using `first_including_muted`, since that's the oldest message ID for the channel that we have in cache?

Reply:
Fixed it in second commit.